cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1020,10 +1020,16 @@ def get_character_states(
             The full character state array of the specified node.
 
         Raises:
-            CassiopeiaTreeError if the tree has not been initialized.
+            CassiopeiaTreeError if the tree has not been initialized or if
+            the character states are not annotated at the specified node.
         """
         self.__check_network_initialized()
 
+        if "character_states" not in self.__network.nodes[node]:
+            raise CassiopeiaTreeError(
+                f"Character states not annotated at node {node}."
+            )
+
```

Comment:
When would this possibly happen?

Any time you attempt to add a tree to a `CassiopeiaTree` object, you'll have to run `populate_tree`, which will by default add character sates to the attributes of each node. If no character states are specified, then an empty list will be used.

I'm not sure that this will ever be hit - can you specify a case where this might happen?

Reply:
This was a bug indeed. If you specify the tree upon initialization but not the character matrix, some nodes would not have the "character_states" attribute initialized (see the first case in `test_states_no_annotation`)  but now I realize with the help of this comment that it's caused be the fact that we're only annotating the attribute as an empty list for internal nodes and not leaf nodes. We should have it do all nodes and it would fix this
![image](https://user-images.githubusercontent.com/29552136/128781796-25545681-e95f-4839-bcd0-281f89b35819.png)